diff --binary -ruN qt/qtbase/mkspecs/win32-g++/qmake.conf qt2/qtbase/mkspecs/win32-g++/qmake.conf
--- qt/qtbase/mkspecs/win32-g++/qmake.conf	2014-09-11 14:48:07.000000000 +0400
+++ qt2/qtbase/mkspecs/win32-g++/qmake.conf	2014-11-27 00:31:19.680802100 +0300
@@ -25,11 +25,11 @@
 QMAKE_LEXFLAGS          =
 QMAKE_YACC              = byacc
 QMAKE_YACCFLAGS         = -d
-QMAKE_CFLAGS            = -pipe -fno-keep-inline-dllexport
+QMAKE_CFLAGS            = -pipe -fno-keep-inline-dllexport -mtune=generic -mmmx -msse -msse2 -fstack-protector-all
 QMAKE_CFLAGS_DEPS       = -M
 QMAKE_CFLAGS_WARN_ON    = -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF   = -w
-QMAKE_CFLAGS_RELEASE    = -O2
+QMAKE_CFLAGS_RELEASE    = -O2 -fomit-frame-pointer -fpredictive-commoning -finline-small-functions
 QMAKE_CFLAGS_DEBUG      = -g
 QMAKE_CFLAGS_YACC       = -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_SPLIT_SECTIONS = -ffunction-sections
@@ -68,7 +68,7 @@
 
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc
-QMAKE_LFLAGS            =
+QMAKE_LFLAGS            = -fstack-protector-all -Wl,-O1 -Wl,--as-needed -Wl,--nxcompat -Wl,--dynamicbase
 QMAKE_LFLAGS_EXCEPTIONS_ON = -mthreads
 QMAKE_LFLAGS_EXCEPTIONS_OFF =
 QMAKE_LFLAGS_RELEASE    = -Wl,-s
diff --binary -ruN qt/qtbase/qmake/Makefile.unix qt2/qtbase/qmake/Makefile.unix
--- qt/qtbase/qmake/Makefile.unix	2014-09-11 14:48:00.000000000 +0400
+++ qt2/qtbase/qmake/Makefile.unix	2014-11-27 00:43:12.173663100 +0300
@@ -101,7 +101,8 @@
 	   -DQT_BUILD_QMAKE -DQT_BOOTSTRAPPED -DPROEVALUATOR_FULL \
 	   -DQT_NO_TEXTCODEC -DQT_NO_UNICODETABLES -DQT_NO_COMPONENT -DQT_NO_COMPRESS \
 	   -DQT_NO_THREAD -DQT_NO_QOBJECT -DQT_NO_GEOM_VARIANT -DQT_NO_DATASTREAM \
-	   -DQT_CRYPTOGRAPHICHASH_ONLY_SHA1 -DQT_JSON_READONLY
+	   -DQT_CRYPTOGRAPHICHASH_ONLY_SHA1 -DQT_JSON_READONLY -mtune=generic -mmmx -msse -msse2 -pipe -fstack-protector-all \
+	   -O3 -fomit-frame-pointer -fpredictive-commoning -finline-small-functions
 
 CXXFLAGS = $(EXTRA_CXXFLAGS) $(CPPFLAGS)
 
@@ -109,7 +110,7 @@
 qmake: $(BUILD_PATH)/bin/qmake$(EXEEXT)
 
 $(BUILD_PATH)/bin/qmake$(EXEEXT): $(OBJS) $(QOBJS)
-	$(CXX) -o "$@" $(OBJS) $(QOBJS) $(LFLAGS)
+	$(CXX) -o "$@" $(OBJS) $(QOBJS) $(LFLAGS) -fstack-protector-all -Wl,-s -Wl,-O1 -Wl,--as-needed -Wl,--nxcompat -Wl,--dynamicbase
 
 Makefile: $(SOURCE_PATH)/qmake/Makefile.unix
 	@echo "Out of date, please rerun configure"
