diff -ruN Qt/mkspecs/win32-g++/qmake.conf Qt2/mkspecs/win32-g++/qmake.conf
--- Qt/mkspecs/win32-g++/qmake.conf	2014-04-10 22:37:12.000000000 +0400
+++ Qt2/mkspecs/win32-g++/qmake.conf	2014-12-21 01:26:58.155922800 +0300
@@ -27,11 +27,11 @@
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= byacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -pipe -mtune=generic -mmmx -msse -msse2 -fstack-protector-all
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2
+QMAKE_CFLAGS_RELEASE	= -O2 -fomit-frame-pointer -fpredictive-commoning -finline-small-functions
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
@@ -60,7 +60,7 @@
 
 QMAKE_LINK		= $${CROSS_COMPILE}g++
 QMAKE_LINK_C		= $${CROSS_COMPILE}gcc
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		= -fstack-protector-all -Wl,-O1 -Wl,--as-needed -Wl,--nxcompat -Wl,--dynamicbase
 QMAKE_LFLAGS_EXCEPTIONS_ON = -mthreads
 QMAKE_LFLAGS_EXCEPTIONS_OFF =
 QMAKE_LFLAGS_RELEASE	= -Wl,-s
diff -ruN Qt/qmake/Makefile.win32-g++ Qt2/qmake/Makefile.win32-g++
--- Qt/qmake/Makefile.win32-g++	2014-04-10 22:37:13.000000000 +0400
+++ Qt2/qmake/Makefile.win32-g++	2014-12-21 01:28:01.000973800 +0300
@@ -27,7 +27,7 @@
 		-DQT_BUILD_QMAKE -DQT_NO_THREAD -DQT_NO_QOBJECT -DQT_NO_GEOM_VARIANT -DQT_NO_DATASTREAM \
 		-DQT_BOOTSTRAPPED -DQLIBRARYINFO_EPOCROOT
 CXXFLAGS    =   $(CFLAGS)
-LFLAGS      = -static-libgcc -s
+LFLAGS      = -static-libgcc -s -fstack-protector-all -Wl,-s -Wl,-O1 -Wl,--as-needed -Wl,--nxcompat -Wl,--dynamicbase
 LIBS	    = -lole32 -luuid -ladvapi32 -lkernel32
 LINKQMAKE   =	g++ $(LFLAGS) -o qmake.exe $(OBJS) $(QTOBJS) $(LIBS)
 ADDCLEAN    =
@@ -42,7 +42,8 @@
               symmake_abld.o symmake_sbsv2.o symbiancommon.o registry.o epocroot.o gbuild.o
 
 ifdef QMAKE_OPENSOURCE_EDITION
-CFLAGS      += -DQMAKE_OPENSOURCE_EDITION
+CFLAGS      += -DQMAKE_OPENSOURCE_EDITION -mtune=generic -mmmx -msse -msse2 -pipe -fstack-protector-all \
+				-O3 -fomit-frame-pointer -fpredictive-commoning -finline-small-functions
 endif
 
 #qt code
