diff -ruN Qt/qtwebkit/Source/WebCore/platform/text/TextEncodingRegistry.cpp Qt2/qtwebkit/Source/WebCore/platform/text/TextEncodingRegistry.cpp
--- Qt/qtwebkit/Source/WebCore/platform/text/TextEncodingRegistry.cpp	2015-02-17 07:57:06.000000000 +0300
+++ Qt2/qtwebkit/Source/WebCore/platform/text/TextEncodingRegistry.cpp	2015-03-09 16:35:28.906143100 +0300
@@ -298,7 +298,7 @@
     TextCodecMac::registerCodecs(addToTextCodecMap);
 #endif
 
-#if OS(WINDOWS) && USE(WCHAR_UNICODE)
+#if OS(WINCE) && USE(WCHAR_UNICODE)
     TextCodecWin::registerExtendedEncodingNames(addToTextEncodingNameMap);
     TextCodecWin::registerExtendedCodecs(addToTextCodecMap);
 #endif
