--- branches/RC_0_16/Jamfile	2013/02/06 06:00:42	7983
+++ branches/RC_0_16/Jamfile	2013/02/06 09:33:51	7984
@@ -305,9 +305,6 @@
 
 feature need-librt : no yes : composite propagated link-incompatible ;
 
-feature boost-date-time : off on : composite propagated link-incompatible ;
-feature.compose <boost-date-time>off : <define>TORRENT_NO_BOOST_DATE_TIME ;
-
 feature fiemap : off on : composite propagated ;
 feature.compose <fiemap>on : <define>HAVE_LINUX_FIEMAP_H ;
 
