--- a/branches/RC_0_16/include/libtorrent/disk_buffer_pool.hpp
+++ b/branches/RC_0_16/include/libtorrent/disk_buffer_pool.hpp
@@ -32,6 +32,8 @@
 
 #ifndef TORRENT_DISK_BUFFER_POOL
 #define TORRENT_DISK_BUFFER_POOL
+
+#include <boost/utility.hpp>
 
 #include "libtorrent/config.hpp"
 #include "libtorrent/thread.hpp"
 
--- a/branches/RC_0_16/src/error_code.cpp
+++ b/branches/RC_0_16/src/error_code.cpp
@@ -40,12 +40,12 @@
 {
 #if BOOST_VERSION >= 103500
 
-	const char* libtorrent_error_category::name() const
+	const char* libtorrent_error_category::name() const throw()
 	{
 		return "libtorrent error";
 	}
 
-	std::string libtorrent_error_category::message(int ev) const
+	std::string libtorrent_error_category::message(int ev) const throw()
 	{
 		static char const* msgs[] =
 		{
@@ -263,12 +263,12 @@
 		return msgs[ev];
 	}
 
-	const char* http_error_category::name() const
+	const char* http_error_category::name() const throw()
 	{
 		return "http error";
 	}
 
-	std::string http_error_category::message(int ev) const
+	std::string http_error_category::message(int ev) const throw()
 	{
 		std::string ret;
 		ret += to_string(ev).elems;
--- a/branches/RC_0_16/include/libtorrent/error_code.hpp
+++ b/branches/RC_0_16/include/libtorrent/error_code.hpp
@@ -329,9 +329,9 @@
 
 	struct TORRENT_EXPORT libtorrent_error_category : boost::system::error_category
 	{
-		virtual const char* name() const;
-		virtual std::string message(int ev) const;
-		virtual boost::system::error_condition default_error_condition(int ev) const
+		virtual const char* name() const throw();
+		virtual std::string message(int ev) const throw();
+		virtual boost::system::error_condition default_error_condition(int ev) const throw()
 		{ return boost::system::error_condition(ev, *this); }
 	};
 
@@ -343,9 +343,9 @@
 
 	struct TORRENT_EXPORT http_error_category : boost::system::error_category
 	{
-		virtual const char* name() const;
-		virtual std::string message(int ev) const;
-		virtual boost::system::error_condition default_error_condition(int ev) const
+		virtual const char* name() const throw();
+		virtual std::string message(int ev) const throw();
+		virtual boost::system::error_condition default_error_condition(int ev) const throw()
 		{ return boost::system::error_condition(ev, *this); }
 	};