diff --binary -ruN 1/src/file.cpp 2/src/file.cpp
--- 1/src/file.cpp	2013-01-14 00:11:45.000000000 +0400
+++ 2/src/file.cpp	2013-01-22 21:09:07.926271800 +0400
@@ -1346,7 +1346,12 @@
 			if (GetOverlappedResult(m_file_handle, &ol, &num_read, true) == 0)
 			{
 				DWORD last_error = GetLastError();
-				if (last_error != ERROR_HANDLE_EOF)
+				if (last_error != ERROR_HANDLE_EOF
+					&& last_error != ERROR_IO_PENDING
+#ifndef TORRENT_MINGW
+					&& last_error != ERROR_CANT_WAIT
+#endif
+)
 				{
 #ifndef TORRENT_MINGW
 					TORRENT_ASSERT(last_error != ERROR_CANT_WAIT);
@@ -1585,12 +1590,19 @@
 			if (GetOverlappedResult(m_file_handle, &ol, &num_written, true) == 0)
 			{
 				DWORD last_error = GetLastError();
+				if (last_error != ERROR_IO_PENDING
 #ifndef TORRENT_MINGW
-				TORRENT_ASSERT(last_error != ERROR_CANT_WAIT);
+					&& last_error != ERROR_CANT_WAIT
 #endif
-				ec.assign(last_error, get_system_category());
-				CloseHandle(ol.hEvent);
-				return -1;
+)
+				{
+#ifndef TORRENT_MINGW
+					TORRENT_ASSERT(last_error != ERROR_CANT_WAIT);
+#endif
+					ec.assign(last_error, get_system_category());
+					CloseHandle(ol.hEvent);
+					return -1;
+				}
 			}
 			if (num_written < ret) ret = num_written;
 		}
