diff -ruN --binary 1/include/libtorrent/escape_string.hpp 2/include/libtorrent/escape_string.hpp
--- 1/include/libtorrent/escape_string.hpp	2012-08-26 18:53:12.000000000 +0400
+++ 2/include/libtorrent/escape_string.hpp	2012-12-24 12:53:19.958646500 +0400
@@ -60,7 +60,7 @@
 	TORRENT_EXTRA_EXPORT std::string base64decode(std::string const& s);
 	// encodes a string using the base32 scheme
 	TORRENT_EXTRA_EXPORT std::string base32encode(std::string const& s);
-	TORRENT_EXTRA_EXPORT std::string base32decode(std::string const& s);
+	TORRENT_EXPORT std::string base32decode(std::string const& s);
 
 	TORRENT_EXTRA_EXPORT std::string url_has_argument(
 		std::string const& url, std::string argument, std::string::size_type* out_pos = 0);
