diff --binary -ruN 1/Jamfile 2/Jamfile
--- 1/Jamfile	2013-05-12 03:02:21.000000000 +0400
+++ 2/Jamfile	2013-07-28 18:10:33.615162600 +0400
@@ -272,7 +272,6 @@
 		# appears to have some types not fully exported)
 		result += <cflags>-fvisibility=hidden ;
 		result += <cxxflags>-fvisibility-inlines-hidden ;
-		result += <linkflags>-Wl,-Bsymbolic ;
 	}
 
 	return $(result) ;
@@ -408,8 +407,8 @@
 feature export-extra : off on : composite propagated ;
 
 # required for openssl on windows
-lib ssleay32 : : <name>ssleay32 ;
-lib libeay32 : : <name>libeay32 ;
+lib ssleay32 : : <name>crypto ;
+lib libeay32 : : <name>ssl ;
 lib advapi32 : : <name>Advapi32 ;
 lib user32 : : <name>User32 ;
 lib shell32 : : <name>shell32 ;
 