--- branches/RC_0_16/src/file.cpp	2013/01/11 04:43:08	7844
+++ branches/RC_0_16/src/file.cpp	2013/01/25 05:54:36	7932
@@ -1342,8 +1342,9 @@
 				CloseHandle(ol.hEvent);
 				return -1;
 			}
+			WaitForSingleObject(ol.hEvent, INFINITE);
 			DWORD num_read;
-			if (GetOverlappedResult(m_file_handle, &ol, &num_read, true) == 0)
+			if (GetOverlappedResult(m_file_handle, &ol, &num_read, false) == 0)
 			{
 				DWORD last_error = GetLastError();
 				if (last_error != ERROR_HANDLE_EOF)
@@ -1581,8 +1582,9 @@
 				CloseHandle(ol.hEvent);
 				return -1;
 			}
+			WaitForSingleObject(ol.hEvent, INFINITE);
 			DWORD num_written;
-			if (GetOverlappedResult(m_file_handle, &ol, &num_written, true) == 0)
+			if (GetOverlappedResult(m_file_handle, &ol, &num_written, false) == 0)
 			{
 				DWORD last_error = GetLastError();
 #ifndef TORRENT_MINGW
