diff --binary -ruN 1/src/src.pro 2/src/src.pro
--- 1/src/src.pro	2012-12-24 13:26:49.071661700 +0400
+++ 2/src/src.pro	2012-12-28 14:08:51.937561900 +0400
@@ -52,10 +52,6 @@
 # Fast concatenation (Qt >= 4.6)
 DEFINES += QT_USE_FAST_CONCATENATION QT_USE_FAST_OPERATOR_PLUS
 
-# Fixes compilation with Boost >= v1.46 where boost
-# filesystem v3 is the default.
-DEFINES += BOOST_FILESYSTEM_VERSION=2
-
 INCLUDEPATH += $$PWD
 
 
diff --binary -ruN 1/winconf-msvc.pri 2/winconf-msvc.pri
--- 1/winconf-msvc.pri	2012-12-22 17:14:25.988984700 +0400
+++ 2/winconf-msvc.pri	2012-12-28 14:44:13.183640000 +0400
@@ -5,11 +5,11 @@
 
 #Adapt the lib names/versions accordingly
 CONFIG(debug, debug|release) {
-  LIBS += libtorrentd.lib \
-          libboost_system-vc90-mt-sgd-1_51.lib
+  LIBS += libtorrent.lib \
+          libboost_system.lib
 } else {
   LIBS += libtorrent.lib \
-          libboost_system-vc90-mt-s-1_51.lib
+          libboost_system.lib
 }
 
 LIBS += advapi32.lib shell32.lib crypt32.lib
diff --binary -ruN 1/winconf.pri 2/winconf.pri
--- 1/winconf.pri	2012-12-22 17:14:25.989984800 +0400
+++ 2/winconf.pri	2012-12-28 15:21:20.732871300 +0400
@@ -1,28 +1,32 @@
 # Adapt these paths on Windows
 
 #Point this to the boost include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/boost_1_51_0)
+INCLUDEPATH += $$quote(T:/_outdir/Boost/Boost64_S/include)
 #Point this to the libtorrent include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/RC_0_16/include)
+INCLUDEPATH += $$quote(T:/_outdir/libtorrent/libtorrent64_S/include)
+#OpenSSL include path
+INCLUDEPATH += $$quote(T:/_outdir/OpenSSL/OpenSSL64_S/include)
 
 #Point this to the boost lib folder
-LIBS += $$quote(-LC:/qBittorrent/boost_1_51_0/stage/lib)
+LIBS += $$quote(-LT:/_outdir/Boost/Boost64_S/lib)
 #Point this to the libtorrent lib folder
-LIBS += $$quote(-LC:/qBittorrent/RC_0_16/bin/<path-according-to-the-build-options-chosen>)
+LIBS += $$quote(-LT:/_outdir/libtorrent/libtorrent64_S/lib)
+#OpenSSL lib path
+LIBS += $$quote(-LT:/_outdir/OpenSSL/OpenSSL64_S/lib)
 
 # LIBTORRENT DEFINES
 DEFINES += BOOST_ALL_NO_LIB
+DEFINES += BOOST_ASIO_ENABLE_CANCELIO
 DEFINES += BOOST_ASIO_HASH_MAP_BUCKETS=1021
 DEFINES += BOOST_ASIO_SEPARATE_COMPILATION
 DEFINES += BOOST_EXCEPTION_DISABLE
-DEFINES += BOOST_SYSTEM_STATIC_LINK=1
 DEFINES += TORRENT_USE_OPENSSL
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
 DEFINES += _WIN32
 DEFINES += WIN32_LEAN_AND_MEAN
-DEFINES += _WIN32_WINNT=0x0500
+DEFINES += _WIN32_WINNT=0x0600
 DEFINES += _WIN32_IE=0x0500
 DEFINES += _CRT_SECURE_NO_DEPRECATE
 DEFINES += _SCL_SECURE_NO_DEPRECATE
