diff -ru Qt/src/winmain/Makefile.Release Qt2/src/winmain/Makefile.Release
--- Qt/src/winmain/Makefile.Release	2012-06-29 16:07:30.708911400 +0400
+++ Qt2/src/winmain/Makefile.Release	2012-06-29 16:08:11.803261800 +0400
@@ -212,7 +212,7 @@
 install_target: first FORCE
 	@$(CHK_DIR_EXISTS) t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib $(MKDIR) t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib 
 	-$(INSTALL_FILE) "..\..\lib\qtmain.prl" "t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib\qtmain.prl"
-	@$(CHK_DIR_EXISTS) t:$(INSTALL_ROOT)/_outdir/Qt/Qt64_qbt_G/lib/pkgconfig $(MKDIR) t:$(INSTALL_ROOT)/_outdir/Qt/Qt64_qbt_G/lib/pkgconfig 
+	@$(CHK_DIR_EXISTS) t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib\pkgconfig $(MKDIR) t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib\pkgconfig 
 	-$(INSTALL_FILE) "..\..\lib\pkgconfig\qtmain.pc" "t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib\pkgconfig\qtmain.pc"
 	-$(INSTALL_FILE) "$(DESTDIR_TARGET)" "t:$(INSTALL_ROOT)\_outdir\Qt\Qt64_qbt_G\lib\$(TARGET)"
    