diff -ruN qBittorrent_orig/winconf-msvc.pri qBittorrent/winconf-msvc.pri
--- qBittorrent_orig/winconf-msvc.pri	2015-08-02 17:45:47.063880200 +0300
+++ qBittorrent/winconf-msvc.pri	2015-08-02 19:24:18.614879100 +0300
@@ -20,11 +20,11 @@
 
 # Adapt the lib names/versions accordingly
 CONFIG(debug, debug|release) {
-  LIBS += libtorrentd.lib \
-          libboost_system-vc90-mt-sgd-1_51.lib
+  LIBS += torrent.lib \
+          boost_system.lib
 } else {
-  LIBS += libtorrent.lib \
-          libboost_system-vc90-mt-s-1_51.lib
+  LIBS += torrent.lib \
+          boost_system.lib
 }
 
 LIBS += advapi32.lib shell32.lib crypt32.lib User32.lib
diff -ruN qBittorrent_orig/winconf.pri qBittorrent/winconf.pri
--- qBittorrent_orig/winconf.pri	2015-08-02 19:19:24.727543600 +0300
+++ qBittorrent/winconf.pri	2015-08-02 19:25:20.670431700 +0300
@@ -20,17 +20,19 @@
 
 # LIBTORRENT DEFINES
 DEFINES += BOOST_ALL_NO_LIB
+DEFINES += BOOST_ASIO_ENABLE_CANCELIO
 DEFINES += BOOST_ASIO_HASH_MAP_BUCKETS=1021
-DEFINES += BOOST_ASIO_SEPARATE_COMPILATION
+DEFINES += BOOST_ASIO_DYN_LINK
+DEFINES += BOOST_ALL_DYN_LINK
 DEFINES += BOOST_EXCEPTION_DISABLE
-DEFINES += BOOST_SYSTEM_STATIC_LINK=1
 DEFINES += TORRENT_USE_OPENSSL
+DEFINES += TORRENT_LINKING_SHARED
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
 DEFINES += _WIN32
 DEFINES += WIN32_LEAN_AND_MEAN
-DEFINES += _WIN32_WINNT=0x0500
+DEFINES += _WIN32_WINNT=0x0600
 DEFINES += _WIN32_IE=0x0500
 DEFINES += _CRT_SECURE_NO_DEPRECATE
 DEFINES += _SCL_SECURE_NO_DEPRECATE
