diff -ruN --binary 1/src/src.pro 2/src/src.pro
--- 1/src/src.pro	2013-10-04 19:12:47.271873700 +0400
+++ 2/src/src.pro	2013-10-04 19:26:40.363098200 +0400
@@ -58,10 +58,6 @@
 # Fast concatenation (Qt >= 4.6)
 DEFINES += QT_USE_FAST_CONCATENATION QT_USE_FAST_OPERATOR_PLUS
 
-# Fixes compilation with Boost >= v1.46 where boost
-# filesystem v3 is the default.
-#DEFINES += BOOST_FILESYSTEM_VERSION=2
-
 INCLUDEPATH += $$PWD
 
 
diff -ruN --binary 1/winconf-msvc.pri 2/winconf-msvc.pri
--- 1/winconf-msvc.pri	2013-10-04 19:12:37.644323000 +0400
+++ 2/winconf-msvc.pri	2013-10-04 19:27:15.658146900 +0400
@@ -20,11 +20,11 @@
 
 #Adapt the lib names/versions accordingly
 CONFIG(debug, debug|release) {
-  LIBS += libtorrentd.lib \
-          libboost_system-vc90-mt-sgd-1_51.lib
+  LIBS += torrent.lib \
+          boost_system.lib
 } else {
-  LIBS += libtorrent.lib \
-          libboost_system-vc90-mt-s-1_51.lib
+  LIBS += torrent.lib \
+          boost_system.lib
 }
 
 LIBS += advapi32.lib shell32.lib crypt32.lib User32.lib
diff -ruN --binary 1/winconf.pri 2/winconf.pri
--- 1/winconf.pri	2013-10-04 01:46:35.949154800 +0400
+++ 2/winconf.pri	2013-10-04 19:29:17.603315200 +0400
@@ -1,32 +1,38 @@
 # Adapt these paths on Windows
 
 #Point this to the boost include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/boost_1_51_0)
+INCLUDEPATH += $$quote(C:/_/_outdir/Boost/Boost64/include)
 #Point this to the libtorrent include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/RC_0_16/include)
+INCLUDEPATH += $$quote(C:/_/_outdir/libtorrent/libtorrent64/include)
 #Point this to the zlib include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/Zlib/include)
+INCLUDEPATH += $$quote(C:/_/_outdir/Zlib/Zlib64/include)
+#OpenSSL include path
+INCLUDEPATH += $$quote(C:/_/_outdir/OpenSSL/OpenSSL64/include)
 
 #Point this to the boost lib folder
-LIBS += $$quote(-LC:/qBittorrent/boost_1_51_0/stage/lib)
+LIBS += $$quote(-LC:/_/_outdir/Boost/Boost64/lib)
 #Point this to the libtorrent lib folder
-LIBS += $$quote(-LC:/qBittorrent/RC_0_16/bin/<path-according-to-the-build-options-chosen>)
+LIBS += $$quote(-LC:/_/_outdir/libtorrent/libtorrent64/lib)
 #Point this to the zlib lib folder
-LIBS += $$quote(-LC:/qBittorrent/Zlib/lib)
+LIBS += $$quote(-LC:/_/_outdir/Zlib/Zlib64/lib)
+#OpenSSL lib path
+LIBS += $$quote(-LC:/_/_outdir/OpenSSL/OpenSSL64/lib)
 
 # LIBTORRENT DEFINES
 DEFINES += BOOST_ALL_NO_LIB
+DEFINES += BOOST_ASIO_ENABLE_CANCELIO
 DEFINES += BOOST_ASIO_HASH_MAP_BUCKETS=1021
-DEFINES += BOOST_ASIO_SEPARATE_COMPILATION
+DEFINES += BOOST_ASIO_DYN_LINK
+DEFINES += BOOST_ALL_DYN_LINK
 DEFINES += BOOST_EXCEPTION_DISABLE
-DEFINES += BOOST_SYSTEM_STATIC_LINK=1
 DEFINES += TORRENT_USE_OPENSSL
+DEFINES += TORRENT_LINKING_SHARED
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
 DEFINES += _WIN32
 DEFINES += WIN32_LEAN_AND_MEAN
-DEFINES += _WIN32_WINNT=0x0500
+DEFINES += _WIN32_WINNT=0x0600
 DEFINES += _WIN32_IE=0x0500
 DEFINES += _CRT_SECURE_NO_DEPRECATE
 DEFINES += _SCL_SECURE_NO_DEPRECATE
