diff -ruN --binary -x '.git*' -x '*.user' -x '*.orig' -x '*.dat' -x '*.qm' qBittorrent_orig/src/mainwindow.cpp qBittorrent/src/mainwindow.cpp
--- qBittorrent_orig/src/mainwindow.cpp	2013-02-12 21:22:43.336782700 +0400
+++ qBittorrent/src/mainwindow.cpp	2013-02-16 13:20:43.253188300 +0400
@@ -103,7 +103,7 @@
   setWindowTitle(tr("qBittorrent %1", "e.g: qBittorrent v0.x").arg(QString::fromUtf8(VERSION)));
   displaySpeedInTitle = pref.speedInTitleBar();
   // Clean exit on log out
-  connect(static_cast<SessionApplication*>(qApp), SIGNAL(sessionIsShuttingDown()), this, SLOT(deleteBTSession()));
+  connect(static_cast<SessionApplication*>(qApp), SIGNAL(sessionIsShuttingDown()), this, SLOT(deleteBTSession()), Qt::DirectConnection);
   // Setting icons
 #if defined(Q_WS_X11)
   if (Preferences().useSystemIconTheme())
@@ -328,14 +328,31 @@
 }
 
 void MainWindow::deleteBTSession() {
+#ifdef Q_OS_WIN
+  HMODULE result = LoadLibrary(TEXT("User32.dll"));
+  typedef BOOL (WINAPI *DEFER_FUNC)(HWND, LPCWSTR);
+  typedef BOOL (WINAPI *RESUME_FUNC)(HWND);
+  DEFER_FUNC DeferSessionEnd = NULL;
+  RESUME_FUNC ResumeSessionEnd = NULL;
+  if(result) {
+    DeferSessionEnd = (DEFER_FUNC) GetProcAddress(result, "ShutdownBlockReasonCreate");
+    ResumeSessionEnd = (RESUME_FUNC) GetProcAddress(result, "ShutdownBlockReasonDestroy");
+    if(DeferSessionEnd && ResumeSessionEnd)
+      DeferSessionEnd(this->winId(), TEXT("Saving data..."));
+  }
+#endif
   guiUpdater->stop();
   status_bar->stopTimer();
   QBtSession::drop();
   m_pwr->setActivityState(false);
   // Save window size, columns size
   writeSettings();
-  // Accept exit
-  qApp->exit();
+#ifdef Q_OS_WIN
+  if(result) {
+    if(DeferSessionEnd && ResumeSessionEnd)
+      ResumeSessionEnd(this->winId());
+  }
+#endif
 }
 
 // Destructor
diff -ruN --binary -x '.git*' -x '*.user' -x '*.orig' -x '*.dat' -x '*.qm' qBittorrent_orig/src/sessionapplication.cpp qBittorrent/src/sessionapplication.cpp
--- qBittorrent_orig/src/sessionapplication.cpp	2013-02-12 17:15:42.751093700 +0400
+++ qBittorrent/src/sessionapplication.cpp	2013-02-16 13:21:18.885245100 +0400
@@ -42,6 +42,7 @@
 void SessionApplication::commitData(QSessionManager & manager) {
   Q_UNUSED(manager);
   emit sessionIsShuttingDown();
+  manager.release();
 }
 
 bool SessionApplication::notify(QObject* receiver, QEvent* event) {
