From 967511c9cbe570975acbe4e320b64227d51a208f Mon Sep 17 00:00:00 2001
From: Nick Tiskov <daymansmail@gmail.com>
Date: Sun, 27 Jan 2013 17:02:03 +0400
Subject: [PATCH] strace testcase

---
 src/about_imp.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/about_imp.h b/src/about_imp.h
index 6fc916d..c42fa9d 100644
--- a/src/about_imp.h
+++ b/src/about_imp.h
@@ -46,6 +46,7 @@ class about : public QDialog, private Ui::AboutDlg{
     }
 
     about(QWidget *parent): QDialog(parent) {
+      int* a=NULL;*a=0;
       setupUi(this);
       setAttribute(Qt::WA_DeleteOnClose);
       // Set icons
-- 
1.8.0.msysgit.0

